File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    const n = arr.length;
    const m = arr[0].length;
    
    if (n > m) {
        for (let i = 0; i < n; i++) {
            for (let j = 0; j < n - m; j++) {
                arr[i].push(0);
            }
        }
    } else {
        for (let i = 0; i < m - n; i++) {
            arr.push(Array(m).fill(0));
        }
    }

    return arr;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    R = len(arr)
    C = len(arr[0])
    
    if R > C:
        num_zeros_to_add = R - C

        for row in arr:
            row.extend([0] * num_zeros_to_add)

    elif C > R:
        num_rows_to_add = C - R

        new_row = [0] * C

        for _ in range(num_rows_to_add):
            arr.append(new_row)

    return arr